dvc/remote/local/__init__.py

```diff
@@ -566,3 +601,73 @@ def unprotect(path_info):
     @staticmethod
     def protect(path_info):
         os.chmod(path_info.path, stat.S_IREAD | stat.S_IRGRP | stat.S_IROTH)
+
+    @classmethod
+    def _append_unpacked_suffix(cls, string):
+        return string + cls.UNPACKED_DIR_SUFFIX
+
+    def _get_unpacked_dir_path_info(self, checksum):
+        unpacked_dir_info = self.checksum_to_path_info(checksum)
+        unpacked_dir_info.path = self._append_unpacked_suffix(
+            unpacked_dir_info.path
+        )
+
+        return unpacked_dir_info
+
+    def _path_info_changed(self, path_info):
+        if self.exists(path_info) and self.state.get(path_info):
+            return False
+        return True
+
+    def _update_unpacked_dir(self, checksum, progress_callback=None):
+        unpacked_dir_info = self._get_unpacked_dir_path_info(checksum)
+
+        if not self._path_info_changed(unpacked_dir_info):
+            return
+
+        self.remove(unpacked_dir_info)
+
+        try:
+            dir_info = self.get_dir_cache(checksum)
+            self._create_unpacked_dir(
+                checksum, dir_info, unpacked_dir_info, progress_callback
+            )
+        except Exception:
```

Comment:
This is too generic.

Reply:
Agreed, but in this particular case I think it is good idea to catch generic exception. I don't think that there exist situation where we want to fail `checkout` or `status` because our unpacked dir could not be created. DVC can still work without this optimization, so just logging seems like valid approach to me.

Reply:
@pared just catch something specific. What could be the cause exception? DvcException?

Reply:
@efiop Yes, it seems that hardlink can raise DvcException.